src/app/clusters/water-heater-management-server/water-heater-management-server.cpp

```diff
@@ -0,0 +1,196 @@
+    return CHIP_NO_ERROR;
+}
+
+// CommandHandlerInterface
+CHIP_ERROR Instance::EnumerateAcceptedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback, void * context)
+{
+    using namespace Commands;
+
+    VerifyOrExit(callback(Boost::Id, context) == Loop::Continue, /**/);
+    VerifyOrExit(callback(CancelBoost::Id, context) == Loop::Continue, /**/);
+
+exit:
+    return CHIP_NO_ERROR;
+}
+
+void Instance::InvokeCommand(HandlerContext & handlerContext)
+{
+    using namespace Commands;
+
+    switch (handlerContext.mRequestPath.mCommandId)
+    {
+    case Boost::Id:
+        HandleCommand<Boost::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleBoost(ctx, commandData); });
+        return;
+    case CancelBoost::Id:
+        HandleCommand<CancelBoost::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleCancelBoost(ctx, commandData); });
+        return;
+    }
+}
+
+void Instance::HandleBoost(HandlerContext & ctx, const Commands::Boost::DecodableType & commandData)
+{
+    uint32_t duration                        = commandData.duration;
+    Optional<bool> oneShot                   = commandData.oneShot;
+    Optional<bool> emergencyBoost            = commandData.emergencyBoost;
+    Optional<int16_t> temporarySetpoint      = commandData.temporarySetpoint;
+    Optional<chip::Percent> targetPercentage = commandData.targetPercentage;
+    Optional<chip::Percent> targetReheat     = commandData.targetReheat;
+
+    //  Notify the appliance if the appliance hardware cannot be adjusted, then return Failure
+    if (!HasFeature(WaterHeaterManagement::Feature::kTankPercent))
+    {
+        if (targetPercentage.HasValue())
+        {
+            if (targetPercentage.Value() > 100)
+            {
+                ChipLogError(Zcl, "Bad targetPercentage %u", targetPercentage.Value());
+                ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+                return;
+            }
+        }
+
+        if (targetReheat.HasValue())
+        {
+            if (targetReheat.Value() > 100)
+            {
+                ChipLogError(Zcl, "Bad targetReheat %u", targetReheat.Value());
+                ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+                return;
+            }
+
+            if (!targetPercentage.HasValue())
+            {
+                ChipLogError(Zcl, "targetPercentage must be specified if targetReheat specified");
+                ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+                return;
+            }
+
+            if (oneShot.HasValue())
+            {
+                ChipLogError(Zcl, "Cannot specify targetReheat with targetPercentage and oneShot. oneShot must be excluded");
+                ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
```

Comment:
Spec should define the behavior here, but Failure is probably not the right code.  InvalidCommand is more likely.

Reply:
Done and spec updated to be clearer https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/10028/